lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -490,14 +482,16 @@ def _quoted_partman_schema
   end
 
   def _fully_qualified_table_name_for_partman(table)
-    _schema_and_table_for(table).each do |identifier|
+    table = _struct_from_table_name(table)
+
+    [table.schema, table.name].each do |identifier|
       if identifier.to_s !~ /^[a-z_][a-z_\d]*$/
         raise PgHaMigrations::InvalidMigrationError, "Partman requires schema / table names to be lowercase with underscores"
       end
     end.join(".")
   end
 
-  def _schema_and_table_for(table)
+  def _struct_from_table_name(table)
```

Comment:
I think we should move this code directly into `Struct::Table.from_table_name`.

Reply:
Related to my comment above (https://github.com/braintree/pg_ha_migrations/pull/87#discussion_r1324759337), I was trying to draw a line with the struct objects such that they only do light validation / formatting and don't actually reach out to the database. If we want to cross that line, I think maybe we'd want to move some of the other helper methods as well like `partitioned?`, `child_partitions`, and `valid?`

Reply:
Update: to hell with the line. This feels pretty clean: https://github.com/braintree/pg_ha_migrations/pull/87/commits/8c05071a751fc560ed02b730f011c22f155527f3

Reply:
Yeah, that seems a lot cleaner.

I wasn't going to mention it, but since you brought it up, I don't think the `Struct` nested module adds anything.

Reply:
Addressed in https://github.com/braintree/pg_ha_migrations/pull/87/commits/bf4407d629c2b4eb8e5f1d23f85f9a8087e8fd06